akka-actor-typed/src/main/scala/akka/actor/typed/pubsub/Topic.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.pubsub
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.internal.pubsub.TopicImpl
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.DoNotInherit
+
+import scala.reflect.ClassTag
+
+/**
+ * A pub sub topic is an actor that handles subscribing to a topic and publishing messages to all subscribed actors.
+ *
+ * It is mostly useful in a clustered setting, where it is intended to be started once on every node that want to
+ * house subscribers or publish messages to the topic, but it also works in a local setting without cluster.
+ *
+ * In a clustered context messages are deduplicated so that there is at most one message sent to each node for
+ * each publish and if there are no subscribers on a node, no message is sent to it. Note that the list of subscribers
+ * is eventually consistent and there are no delivery guarantees built in.
+ *
+ * Each topic results in a [[akka.actor.typed.receptionist.ServiceKey]] in the [[akka.actor.typed.receptionist.Receptionist]]
+ * so the same scaling recommendation holds for topics, see docs:
+ * https://doc.akka.io/docs/akka/current/typed/actor-discovery.html#receptionist-scalability
+ */
+object Topic {
+
+  /**
+   * Not for user extension
+   */
+  @DoNotInherit
+  sealed trait Command[T] extends TopicImpl.Command[T]
+
+  /**
+   * Publish the message to all currently known subscribers.
+   */
+  final case class Publish[T](message: T) extends Command[T]
+
+  /**
+   * Subscribe to this topic. Should only be used for local subscribers.
+   */
+  final case class Subscribe[T](subscriber: ActorRef[T]) extends Command[T]
+
+  /**
+   * Unsubscribe a previously subscribed actor from this topic.
+   */
+  final case class Unsubscribe[T](subscriber: ActorRef[T]) extends Command[T]
```

Comment:
Perhaps acks for subscribe and unsubscribe would be useful? Since all messages go through the topic, we can guarantee ordering of sub-ack published unsub-ack and that no more messages are published after unsub.

Reply:
Are we dropping those Acks from the classic usage, just thinking about existing users migration.

Reply:
I think we need to motivate why they make sense before we add them here as well, for unsub-ack perhaps it would allow you to make sure there are no inflight published that you'd loose but otoh since there is no delivery guarantees anyways perhaps that isn't as useful as it sounds at first.

Reply:
it's always a local subscribe so the ack is not worth much? for the Receptionist I think we made it optional

Reply:
Yeah, let's skip it for now.